pkg/apis/contrail/v1alpha1/swiftproxy_types.go

```diff
@@ -78,12 +73,32 @@ type SwiftProxyList struct {
 // SwiftProxyInstanceType is type unique name used for labels
 const SwiftProxyInstanceType = "SwiftProxy"
 
-//PodsCertSubjects gets list of SwiftProxy pods certificate subjets which can be passed to the certificate API
+// PodsCertSubjects gets list of SwiftProxy pods certificate subjets which can be passed to the certificate API
 func (s *SwiftProxy) PodsCertSubjects(podList *corev1.PodList, serviceIP string) []certificates.CertificateSubject {
 	altIPs := PodAlternativeIPs{ServiceIP: serviceIP}
 	return PodsCertSubjects(podList, s.Spec.CommonConfiguration.HostNetwork, altIPs)
 }
 
+// GetServiceType returns chosen Service type for Swift Proxy, default is LoadBalancer
+func (s *SwiftProxy) GetServiceType() corev1.ServiceType {
+	if s.Spec.ServiceConfiguration.SwiftServiceType == "" {
+		return corev1.ServiceTypeLoadBalancer
+	}
+	return corev1.ServiceType(s.Spec.ServiceConfiguration.SwiftServiceType)
+}
+
+// GetServiceAnnotations prepares annotations for LoadBalancer Service
+func (s *SwiftProxy) GetServiceAnnotations() map[string]string {
+	if s.GetServiceType() == corev1.ServiceTypeLoadBalancer {
+		return map[string]string{}
+	}
+	poolName := "mgmt"
+	if s.Spec.ServiceConfiguration.IPPoolName != "" {
+		poolName = s.Spec.ServiceConfiguration.IPPoolName
+	}
+	return map[string]string{"metallb.universe.tf/address-pool": poolName}
```

Comment:
Please add TODO to remove this default value

Reply:
done